resources/scripts/ttx_diff.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+"""Helper for comparing fontc (Rust) vs fontmake (Python) binaries.
+
+Turns each into ttx, eliminates expected sources of difference and prints
+a brief summary of the result.
+
+fontmake should be installed in an active virtual environment.
+
+Usage:
+    # rebuild with fontmake and fontc and compare
+    python resources/scripts/ttx_diff.py ../OswaldFont/sources/Oswald.glyphs
+    # rebuild the fontc copy, reuse the prior fontmake copy (if present), and compare
+    python resources/scripts/ttx_diff.py --rebuild fontc ../OswaldFont/sources/Oswald.glyphs
+"""
+
+from absl import app
+from absl import flags
+from lxml import etree
+from pathlib import Path
+import shutil
+import subprocess
+import sys
+
+
+FLAGS = flags.FLAGS
+
+
+flags.DEFINE_enum('rebuild', 'both', ['both', 'fontc', 'fontmake'], 'Which compilers to rebuild with if the output appears to already exist')
+
+
+def run(cmd, **kwargs):
+    cmd_string = " ".join(cmd)
+    print(f"Running {cmd_string}")
+    subprocess.run(cmd, text=True, check=True, **kwargs)
+
+
+def ttx(font_file):
+    ttx_file = font_file.with_suffix(".ttx")
+    cmd = [
+        "ttx",
+        "-o",
+        str(ttx_file),
+        str(font_file),
+    ]
+    run(cmd)
+    return ttx_file
+
+
+def build(cmd, build_tool, ttf_find_fn, **kwargs):
+    try_skip = FLAGS.rebuild not in [build_tool, "both"]
+    ttfs = ttf_find_fn()
+    if try_skip and len(ttfs) == 1:
+        ttx_file = ttfs[0].with_suffix(".ttx")
+        if ttx_file.is_file():
+            print(f"skipping {build_tool}")
+            return ttx_file 
+    run(cmd, **kwargs)
+    ttfs = ttf_find_fn()
+    assert len(ttfs) == 1, ttfs
+    return ttx(ttfs[0])
+
+
+def build_fontc(source, build_dir):    
+    cmd = [
+        "cargo",
+        "run",
+        "-p",
+        "fontc",
+        "--",
+        "--source",
+        str(source),
+        "--build-dir",
+        str(build_dir),
+    ]
+    return build(cmd, "fontc", lambda: (build_dir / "font.ttf",))
+
+
+def build_fontmake(source, build_dir):
+    cmd = [
+        "fontmake",
+        "-o",
+        "variable",
+        "--no-production-names",
+        "--keep-direction",
+        "--filter",
+        "FlattenComponentsFilter",
```

Comment:
isn't --flatten_components false by default?

https://github.com/googlefonts/fontmake-rs/blob/f4a9940e2a225a04726d5d923cc01e884d730b54/fontc/src/args.rs#L32

it doesn't look like you have enabled that in the call to fontc. Not saying that you should, just that the fontc vs fontmake calls don't match my expectation given my current understanding as to where we currently stand in regards default options.

Reply:
I wonder if we need to test with/without flatten (for both tools) separately? - we do care it can work in gftools (with it on) and vs default fontmake behavior (with it off)

Reply:
maybe add options to this script to forward to each compiler the relevant filters/options

Reply:
Adding a flag for this, `flags.DEFINE_enum('compare', 'both', ['both', 'default', 'gftools'], 'Compare results with default flags, with the flags gftools uses, or both. Default both.')`